akka-docs/rst/scala/code/docs/stream/KillSwitchDocSpec.scala

```diff
@@ -0,0 +1,106 @@
+package docs.stream
+
+import akka.stream.scaladsl._
+import akka.stream.{ActorMaterializer, DelayOverflowStrategy, KillSwitches}
+import akka.testkit.AkkaSpec
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class KillSwitchDocSpec extends AkkaSpec {
+
+  implicit val materializer = ActorMaterializer()
+
+  "Unique kill switch" must {
+
+    "control graph completion with shutdown" in {
+
+      // format: OFF
+      //#unique-shutdown
+      val countingSrc = Source(Stream.from(1)).delay(1.second, DelayOverflowStrategy.backpressure)
+      val lastSnk = Sink.last[Int]
+
+      val (killSwitch, last) = countingSrc
+        .viaMat(KillSwitches.single)(Keep.right)
+        .toMat(lastSnk)(Keep.both)
+        .run()
+
+      Thread.sleep(2500)
```

Comment:
That is not so nice, and is also adding to the overall test execution time.
Can we just have `doSomethingElse()` and not verify the result in this doc test?